packages/sdk-client-v3/src/middleware/create-concurrent-modification-middleware.ts

```diff
@@ -10,22 +10,23 @@ export default function createConcurrentModificationMiddleware(
     version: number,
     request: MiddlewareRequest,
     response: MiddlewareResponse
-  ) => Promise<Record<string, any> | string | Buffer>
+  ) => Promise<Record<string, any> | string | Uint8Array>
 ): Middleware {
   return (next: Next) => {
     return async (request: MiddlewareRequest): Promise<MiddlewareResponse> => {
+      request['firsAttempt'] = true
       const response = await next(request)
       if (response.statusCode == 409) {
         /**
          * extract the currentVersion
          * from the error body and update
          * request with the currentVersion
          */
-        const version = response.error.body?.errors?.[0]?.currentVersion
+        const version = response.error?.body?.errors?.[0]?.currentVersion
 
         // update the resource version here
         if (version) {
-          if (modifierFunction) {
+          if (modifierFunction && typeof modifierFunction == 'function') {
```

Comment:
Yes, it is important to ensure this is a function (at runtime). First not all customers are using typescript and secondly any argument other than a function can be passed so we are trying to prevent that.